ckanext/canada/helpers.py

```diff
@@ -641,3 +646,84 @@ def flash_success(message, allow_html=True):
+        return ''
+
+    is_datastore_active = resource.get('datastore_active', False)
+
+    try:
+        xloader_job = t.get_action("xloader_status")(None, {"resource_id": resource.get('id')})
+    except (t.ObjectNotFound, t.NotAuthorized):
+        xloader_job = {}
+
+    if xloader_job.get('status') == 'complete':
+        # the xloader task is complete, show datastore active or inactive.
+        # xloader will delete the datastore table at the beggining of the job run.
+        # so this will only be true if the job is fully finished.
+        status = 'active' if is_datastore_active else 'inactive'
+    elif xloader_job.get('status') in ['pending', 'running', 'running_but_viewable', 'error']:
+        # the job is running or pending or errored
+        # show the xloader status
+        status = xloader_job.get('status')
+        if status == 'running_but_viewable':
+            # treat running_but_viewable the same as running
+            status = 'running'
+    else:
+        # we do not know what the status is
+        status = 'unknown'
+
+    messages = {
+        'pending': _('Data awaiting load to DataStore'),
+        'running': _('Loading data into DataStore'),
+        'complete': _('Data loaded into DataStore'),
+        'error': _('Failed to load data into DataStore'),
+        'active': _('Data available in DataStore'),
+        'inactive': _('Resource not active in DataStore'),
+        'unknown': _('DataStore status unknown'),
+    }
+
+    pusher_url = t.h.url_for('xloader.resource_data',
+                             id=resource.get('package_id'),
+                             resource_id=resource.get('id'))
+
+    badge_url = t.h.url_for_static('/static/img/badges/{lang}/datastore-{status}.svg'.format(lang=t.h.lang(), status=status))
+
+    # in queue, try to get the number in the queue
+    # NOTE: the Validation extension does not use the task_status table,
+    # and we will never know the job id of a Validation job.
+    # FIXME: is this worth it if we cannot display in the Xloader page?
+    # is it worth it if we cannot display for the Validation extension?
+    queue_number = None
+    if status == 'pending':
+        queue_number = 1
+        jobs_in_queue = t.get_action('job_list')({"ignore_auth": True}, {})
+        for job in jobs_in_queue:
+            if job.get('id') == xloader_job.get('job_id'):
+                break
+            queue_number += 1
+
+    title = t.h.render_datetime(xloader_job.get('last_updated'), with_hours=True) \
+        if xloader_job.get('last_updated') else ''
+    if queue_number:
+        title = _("Number %s in queue") % queue_number
+
+    # TODO: use str instead of unicode for py3
+    return unicode('<a href="{pusher_url}" class="loader-badge"><img src="{badge_url}" alt="{alt}" title="{title}"/></a>').format(
+        pusher_url=pusher_url,
+        badge_url=badge_url,
+        alt=messages[status],
+        title=title)
```

Comment:
can alt or title ever include `"`? (if so there's html injection issues here)

Reply:
No, that would surely break the html structure. Hmmm for the `alt`, it should be fine because those are static UNLESS gettext returns something else...so yeah we will need to html encode those for sure.

Is there a CKAN way to do that? or just using python libs:
```
try:
    from html import escape  # python 3.x
except ImportError:
    from cgi import escape  # python 2.x
```

Reply:
escaping html with webhelpers that is already in use in this script. And escaping double quotes as well.